streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -221,6 +221,9 @@ State setState(final State newState) {
                 throw new StreamsException(logPrefix + "Unexpected state transition from " + oldState + " to " + newState);
             } else {
                 log.info("State transition from {} to {}", oldState, newState);
+                if (newState == State.DEAD) {
+                    failedStreamThreadSensor.record();
+                }
```

Comment:
Not every dead stream thread is a failed stream thread. You should record this metric where the uncaught exception handler is called because there we now that a stream thread died unexpectedly.

Reply:
Would that just be in `run()` of the GlobalStreamThread then?

Reply:
No, that would be in `StreamThread#runLoop()`.